### PR TITLE
Ataymano/queue size control

### PR DESCRIPTION
**Pull request by ataymano**
Capacity-based queue pruning

### CODE REVIEW

File: examples/test_cpp/experiment_controller.cc [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+void experiment_controller::progress_bar() {
+  if (_iteration % 100 == 0) std::cout << "\r" << _iteration << " sent";
+}
+
+iterations_experiment_controller::iterations_experiment_controller(size_t num_iterations)
+  : _num_iterations(num_iterations)
+{ }
+
+void iterations_experiment_controller::on_iterate()
+{
```
Comment (jackgerrits) [inline examples/test_cpp/experiment_controller.cc]: nit: keep braces consistent

File: examples/test_cpp/experiment_controller.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <boost/program_options.hpp>
```
Comment (jackgerrits) [inline examples/test_cpp/experiment_controller.h]: please add program options as an explicit link dependency to the cmake target

File: examples/test_cpp/experiment_controller.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+class experiment_controller {
+public:
+  virtual ~experiment_controller() {}
+  void restart();
+  void iterate();
+
+  void progress_bar();
+  size_t get_iteration() const;
+
+public:
```
Comment (jackgerrits) [inline examples/test_cpp/experiment_controller.h]: nit: is this `public` needed

File: examples/test_cpp/experiment_controller.h [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  bool is_running() const;
+
+protected:
+  virtual void on_restart() {}
```
Comment (jackgerrits) [inline examples/test_cpp/experiment_controller.h]: Should these be pure virtual instead?

File: rlclientlib/logger/event_queue.h [hunk 5]
```diff
@@ -29,6 +32,7 @@ namespace reinforcement_learning {
       if (!_queue.empty())
       {
         *item = std::move(_queue.front());
+        _capacity = (std::max)(0, static_cast<int>(_capacity) - static_cast<int>(item->size()));
```
Comment (jackgerrits) [inline rlclientlib/logger/event_queue.h]: Why put function name in brackets?

File: rlclientlib/utility/config_utility.cc [hunk 6]
```diff
@@ -1,138 +1,147 @@
...
-        // Otherwise, just set the value in the config collection.
-        cc.set(prop_name.c_str(), string_value.c_str());
-      }
-    }
-
-    return error_code::success;
-  }
-
-}}}
+#include <map>
```
Comment (jackgerrits) [inline rlclientlib/utility/config_utility.cc]: Whats going on with the diff on this file?

Comment (ataymano) [reply rlclientlib/utility/config_utility.cc]: not sure, locally can see proper (small) diff.

File: rlclientlib/logger/event_queue.h (hunk 5 above)
Comment (ataymano) [reply rlclientlib/logger/event_queue.h]: Workaround for disambiguation with windows.h macros.

**Pull request merged**